src/math/trig.c

```diff
@@ -94,9 +122,212 @@ void cordic_sin_cos(int32_t th_rad_fxp, cordic_cfg type, int32_t *sign, int32_t
 		xtmp = *xn >> (b_idx + 1);
 		ytmp = *b_yn >> (b_idx + 1);
 	}
+	/* exponential */
+	if (sign) {
```

Comment:
not sure how sign is, but this could be maybe optimized to multiply by sign or -sign

Reply:
I feel like a bit twiddle solution will be faster than a branch